tasks.py

```diff
@@ -497,44 +515,132 @@ def runStartupTasks():
+		"""
+			Starts iterating the given query on this class. Will return immediately, the first batch will already
+			run deferred.
+
+			Warning: Any custom data *must* be json-serializable and *must* be passed in customData. You cannot store
+			any data on this class as each chunk may run on a different instance!
+		"""
+		assert not (query._customMultiQueryMerge or query._calculateInternalMultiQueryAmount), \
+			"Cannot iter a query with postprocessing"
+		assert isinstance(query.filters, dict), "Unsatisfiable query or query with an IN filter"
+		qryDict = {
+			"kind": query.kind,
+			"srcSkel": query.srcSkel.kindName if query.srcSkel else None,
+			"filters": query.filters,
+			"orders": [(propName, sortOrder.value) for propName, sortOrder in query.orders],
+			"startCursor": query._startCursor,
+			"endCursor": query._endCursor,
+			"origKind": query.origKind,
+			"distinct": query._distinct,
+			"classID": cls.__classID__,
+			"customData": customData,
+			"totalCount": 0
+		}
+		cls._requeueStep(qryDict)
+
+	@classmethod
+	def _requeueStep(cls, qryDict: Dict[str, Any]) -> None:
+		"""
+			Internal use only. Pushes a new step defined in qryDict to either the taskqueue or append it to
+			the current request	if we are on the local development server.
+		"""
+		if not queueRegion: # Run tasks inline - hopefully development server
+			req = currentRequest.get()
+			task = lambda *args, **kwargs: cls._qryStep(qryDict)
+			if req:
+				req.pendingTasks.append(task)  # < This property will be only exist on development server!
+				return
+		project = utils.projectID
+		location = queueRegion
+		parent = taskClient.queue_path(project, location, cls.queueName)
+		task = {
+			'app_engine_http_request': {  # Specify the type of request.
+				'http_method': 'POST',
+				'relative_uri': '/_tasks/queryIter'
+			}
+		}
+		task['app_engine_http_request']['body'] = json.dumps(qryDict, cls=JsonKeyEncoder).encode("UTF-8")
+		taskClient.create_task(parent, task)
+
+	@classmethod
+	def _qryStep(cls, qryDict: Dict[str, Any]) -> None:
+		"""
+			Internal use only. Processes one block of five entries from the query defined in qryDict and
+			reschedules the next block.
+		"""
+		from viur.core.skeleton import skeletonByKind
+		qry = db.Query(qryDict["kind"])
+		qry.srcSkel = skeletonByKind(qryDict["srcSkel"])() if qryDict["srcSkel"] else None
+		qry.filters = qryDict["filters"]
+		qry.orders = [(propName, db.SortOrder(sortOrder)) for propName, sortOrder in qryDict["orders"]]
+		qry._startCursor = qryDict["startCursor"]
+		qry._endCursor = qryDict["endCursor"]
+		qry.origKind = qryDict["origKind"]
+		qry._distinct = qryDict["distinct"]
+		if qry.srcSkel:
+			qryIter = qry.fetch(5)
 		else:
-			sharedConf["viur.disabled"] = False
-"""
+			qryIter = qry.run(5)
+		for item in qryIter:
+			try:
+				cls.handleEntry(item, qryDict["customData"])
+			except:
+				sleep(5)
+				try:
+					cls.handleEntry(item, qryDict["customData"])
+				except Exception as e:
```

Comment:
It could be helpful to have a parameter/option `breakOnException` to stop the whole iteration after an error occurred and call an method like `onError(item, customData, exc)`.